retrieval/raw/raw_endpoint.go

```diff
@@ -43,43 +48,53 @@ func (c *Client) createRequest(method, url string, body io.Reader) (*http.Reques
 }
 
 //GetRawEndPoint retrives the body of HTTP response from a given sourcename, working directory, and URL
-func (c *Client) GetRawEndPoint(sourceName string, workDir *os.File, URL string) (rawRespFile *os.File, rerr error) {
+func (c *Client) GetRawEndPoint(sourceName string, workDir *os.File, URL string, retries uint) (rawRespFile *os.File, rerr error) {
 
 	var empty os.File
 	var fileExt string
+	attempts := retries + 1
 
 	req, err := c.createRequest("GET", URL, nil)
 	if err != nil {
 		return &empty, errors.New("Unable to create raw request for " + sourceName)
 	}
 
-	resp, err := c.HTTPClient.Do(req)
-	if err != nil {
-		return &empty, err
-	}
-
-	defer util.SafeClose(resp.Body.Close, &rerr)
-
-	ct := resp.Header.Get("Content-Type")
-
-	if strings.Contains(ct, "application/json") {
-		fileExt = ".json"
-	} else if strings.Contains(ct, "text/plain") {
-		fileExt = ".txt"
-	} else {
-		fileExt = ""
-	}
-
-	rawRespFile, err = os.Create(workDir.Name() + "/" + sourceName + fileExt)
-	if err != nil {
-		return &empty, errors.New("Unable to create raw metric file")
-	}
-
-	_, err = io.Copy(rawRespFile, resp.Body)
-	if err != nil {
-		fmt.Print(err)
-		return &empty, errors.New("Error writing file: " + rawRespFile.Name())
+	for i := uint(0); i < attempts; i++ {
+		resp, err := c.HTTPClient.Do(req)
+		if err != nil {
+			log.Printf("Unable to connect to URL: %s retrying: %v", URL, i+1)
+			time.Sleep(time.Duration(int64(math.Pow(2, float64(i)))) * time.Second)
+			continue
+		} else if !(resp.StatusCode >= 200 && resp.StatusCode <= 299) {
+			log.Printf("Invalid response from URL: %s retrying: %v", URL, i+1)
+			time.Sleep(time.Duration(int64(math.Pow(2, float64(i)))) * time.Second)
+			continue
+		}
+
+		defer util.SafeClose(resp.Body.Close, &rerr)
```

Comment:
we're technically in a loop here, so doing a defer isn't ideal...even though we don't expect the number of iterations to be high. might consider pulling the body of the for loop into its own method.